yaml.go

```diff
@@ -19,12 +19,24 @@ package yaml
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"fmt"
-	"io"
 	"reflect"
 	"strconv"
 
-	"gopkg.in/yaml.v2"
+	kubejson "sigs.k8s.io/json"
+
+	"gopkg.in/yaml.v3"
+)
+
+type yamlToTargetOption uint8
```

Comment:
this could (for the time being at least) be folded into a `disallowUnknownFields` bool for simplicity and readability